iceoryx_dds/source/iceoryx_dds/dds/cyclone_data_reader.cpp

```diff
@@ -62,14 +62,20 @@ iox::cxx::optional<uint64_t> iox::dds::CycloneDataReader::peekNextSize()
         // Ignore samples with no payload
         if (nextSampleSize != 0)
         {
-            return iox::cxx::optional<uint64_t>(static_cast<uint64_t>(nextSampleSize));
+            return iox::cxx::optional<uint32_t>(static_cast<uint32_t>(nextSampleSize));
```

Comment:
You are right. Why did the signature return value changed from uint64_t to uint32_t?

Reply:
Because `loan()` uses a `uint32_t`:
`cxx::expected<Sample<T>, AllocationError> loan(const uint32_t size) noexcept;`
Used in `inline void DDS2IceoryxGateway<channel_t, gateway_t>::forward(const channel_t& channel) noexcept` in file dd_to_iox.inl.
Should the base pub change instead? @elBoberido @elfenpiff

Reply:
This restriction comes from: 
`cxx::expected<mepoo::ChunkHeader*, AllocationError> tryAllocateChunk(const uint32_t payloadSize) noexcept;`

Reply:
Added to #376